backend/business-partner-agent/src/main/java/org/hyperledger/bpa/impl/aries/ProofManager.java

```diff
@@ -136,6 +153,77 @@ public void sendPresentProofRequest(@NonNull UUID partnerId, @NonNull RequestPro
         }
     }
 
+    public void rejectPresentProofRequest(@NotNull PartnerProof proofEx, String explainString) {
+        if (PresentationExchangeState.REQUEST_RECEIVED.equals(proofEx.getState())) {
+            try {
+                sendPresentProofProblemReport(proofEx.getPresentationExchangeId(), explainString);
+                // after sending rejection notice, delete aries copy. no proper 'reject'
+                // protocol
+                ac.presentProofRecordsRemove(proofEx.getPresentationExchangeId());
+                // after sending removing aries copy, remove BPA copy
+                // TODO: 'rejected_state' to save, so delete for now
+                deletePartnerProof(proofEx.getId());
+            } catch (IOException e) {
+                throw new NetworkException("aca-py not available", e);
+            } catch (AriesException e) {
+                log.error("aca-py wallet item not found, attempting BPA delete");
+            }
+        } else {
+            throw new WrongApiUsageException("PresentationExchangeState != 'request-received'");
+        }
+    }
+
+    public void presentProof(@NotNull PartnerProof proofEx) {
+        if (PresentationExchangeRole.PROVER.equals(proofEx.getRole())
+                && PresentationExchangeState.REQUEST_RECEIVED.equals(proofEx.getState())) {
+            try {
+                ac.presentProofRecordsGetById(proofEx.getPresentationExchangeId())
+                        .ifPresent(record -> presentProof(record));
+            } catch (IOException e) {
+                log.error("aca-py not reachable.", e);
+                return;
+            } catch (AriesException e) {
+                log.error("aca-py wallet item not found");
+            }
+            pProofRepo.updateState(proofEx.getId(), PresentationExchangeState.PRESENTATIONS_SENT);
```

Comment:
Shouldn't the event handler set this? And what happens if above code fails? I would check if this is needed and then either remove or move it in the try section

Reply:
good catch, if the AC call throws an exception, the update will still run. I moved it with the try.